master/buildbot/status/web/status_json.py

```diff
@@ -434,30 +474,132 @@ def asDict(self, request):
 
 
 class BuilderJsonResource(JsonResource):
-    help = """Describe a single builder.
-"""
+    help = 'Describe a single builder.'
     pageTitle = 'Builder'
 
     def __init__(self, status, builder_status):
         JsonResource.__init__(self, status)
         self.builder_status = builder_status
         self.putChild('builds', BuildsJsonResource(status, builder_status))
-        self.putChild('slaves', BuilderSlavesJsonResources(status,
-                                                           builder_status))
-        self.putChild('startslaves', BuilderStartSlavesJsonResources(status,
-                                                           builder_status))
-        self.putChild(
-            'pendingBuilds',
-            BuilderPendingBuildsJsonResource(status, builder_status))
+        self.putChild('slaves', BuilderSlavesJsonResources(status, builder_status))
+        self.putChild('startslaves', BuilderStartSlavesJsonResources(status, builder_status))
+        self.putChild('pendingBuilds', BuilderPendingBuildsJsonResource(status, builder_status))
+        self.putChild('start-build', StartBuildJsonResource(status, builder_status))
 
     def asDict(self, request):
         # buildbot.status.builder.BuilderStatus
-        builder_dict = self.builder_status.asDict_async(codebases=getCodebases(request=request),
-                                                        request=request)
+        builder_dict = self.builder_status.asDict_async(
+            codebases=getCodebases(request=request), request=request,
+        )
 
         return builder_dict
 
 
+class StartBuildJsonResource(AccessorMixin, resource.Resource):
+    help = 'Start a new build'
+    isLeaf = True
+    pageTitle = 'Start a new build'
+    schema = {
+        'type': 'object',
+        'properties': {
+            'force_chain_rebuild': {'type': 'boolean'},
```

Comment:
I guess if the dynamic properties from the force scheduler changed this could get out of sync.

Reply:
These are the default properties, but this endpoint will not accept custom properties for builder, I'll investigate it.